### DISCUSSION TITLE
can't track tables because source is inconsistent

### DISCUSSION THREAD

**Discussion by alilotfy** [hasura/graphql-engine#10235] [Question] [answered]
I am using pg_track_table found in [documentaion](https://hasura.io/docs/latest/api-reference/metadata-api/table-view/) to track some tables. I am always getting the following error: source with name \"Ali_Lotfy\" is inconsistent.

This is the body of the request 
{
    "type": "pg_track_table",
    "args": {
        "source": "Ali_Lotfy",
        "table": "PurchaseOrderDetail",
        "configuration": {
            "custom_name": "PurchaseOrderDetail"
        }
    }
}

How to solve this inconsistency? Is source consistency different from metadata status as the Hasura console is showing metadata status to be consistent?

I want a programmatic way to track tables. If this method won't work, is there another method?

note: I am creating my resources using by sending a post request of type bigquery_add_source.
Tracking tables form the Hasura console is always successful but I want a programmatic method.

@meetzaveri

**Comment by meetzaveri** [comment]
Hello @alilotfy , thanks for reporting. 

I will try to repro this on my end and let you know. Meanwhile, if you can also let me know that on which version of Hasura you are experiencing this on, it would be helpful.

**Reply by alilotfy** [reply]
v2.39.1-cloud.1

**Reply by meetzaveri** [reply]
@alilotfy I was able to track table successfully on my end using [pg_track_table metadata API](https://hasura.io/docs/latest/api-reference/metadata-api/table-view/#metadata-pg-track-table). 

Can you verify if the database(source) named "Ali_Lotfy" exists on your Hasura cloud project ?

**Reply by alilotfy** [reply]
@meetzaveri 
Yes, It does surely exist.
When I type a wrong database name it returns: source with name \"Ali_Lotfy_1\" does not exist
Typing the correct name returns : source with name \"Ali_Lotfy\" is inconsistent

This is the full request 
curl -X POST -H "Content-Type: application/json" -H "X-Hasura-Role: admin" -H "x-hasura-admin-secret:<my-secret>" -d '{
   "type": "pg_track_table",
   "args": {
     "source": "Ali_Lotfy,
     "table": "PurchaseOrderDetail"}}' https://<my_project_name>.hasura.app/v1/metadata
     
Note that I can connect big query using bigquery_add_source. This may be helpful in debugging.
 I suppose the error message means something. What can be the source of inconsistency?

**Reply by alilotfy** [reply]
Now I created a very simple dataset in BigQuery containing one small table with one simple column instead of my actual super complex dataset.
Again I can connect but I can't track tables.

**Reply by meetzaveri** [reply, marked as answer]
Oh, so you are using BQ database and using [bigquery-track-table metadata API](https://hasura.io/docs/latest/api-reference/metadata-api/table-view/#metadata-bigquery-track-table) ? I tried to reproduce using [pg-track-table API](https://hasura.io/docs/latest/api-reference/metadata-api/table-view/#metadata-pg-track-table) earlier.

**Reply by alilotfy** [reply]
Much Thanks. That was my problem I used pg-track-table instead of bigquery-track-table
